### PR TITLE
Feature function library

### PR DESCRIPTION
**Pull request by nikolauska**
Adds requested feature https://github.com/DeKugelschieber/asl/issues/7

My skills with golang are minimal, so there might be be better way to this and some problems I haven't noticed.

### CODE REVIEW

File: src/main/asl.go [hunk 1]
```diff
@@ -105,7 +115,19 @@ func compile(path string) {
     	    fmt.Println("Error writing file: "+aslFiles[i].out)
     	    fmt.Println(err)
     	}
+    }    
+}
+
+func functions(path, functionName string) {
+    functionText := "//class cfgFunctions {\nclass "+functionName+" {\n    class "+functionName+" {\n"
```
Comment (Kugelschieber) [inline src/main/asl.go]: use \r\n to support windows line breaks

**Comment by Kugelschieber** [comment]
I appreciate your help and will look into it later. There might be some things I'd like to change, but I can do this after merging.

**Comment by CaptainPStar** [comment]
I would go a different approach here: compile functions with a keyword like "global" into the function library. This way you can easily mix local and global functions. This can also be a good testbed for classes later down the road.

**Comment by Kugelschieber** [comment]
I agree, so we get a more general approach and no additional compiler flag. Also, as it is currently implemented, the user needs to call asl for each function file and cannot use the recursive option.

**Comment by nikolauska** [comment]
Thanks for the input, that seems like a lot better way. I will redo this when I get back home. 

Just a question. What would be the best way to define function name? For example I like to name my function library like this 

mainName_subName_fnc_function
example kuikka_briefing_fnc_add

**Comment by Kugelschieber** [comment]
Maybe we can define a standardized way for naming. I think of something like:

```
// filename: my_lib.asl
global func myFunc(a, b) {
    // ...
}
```

Output file will be called my_lib_myFunc.sqf
Or just pick the function name itself, so users can integrate whatever style they like.

To implement this the keyword "global" must be added, which is quite easy and can be done in the tokenizer. Exfiltrating the function to its own file is more tricky. I have to experiment to figure out a nice and clean way.

**Comment by nikolauska** [comment]
Ok, well it might be the best I leave this feature for you to implement :)

**Comment by Kugelschieber** [comment]
Ok :)
But you can focus on other features if you like. I really appreciate your help.

**Pull request closed without merging**